test/src/test/java/org/corfudb/runtime/collections/CorfuStoreShimTest.java

```diff
@@ -1168,4 +1172,33 @@ public void checkProtoDeduplication() throws Exception {
         assertThat(numProtoFiles).isEqualTo(numProtoFilesAfterChange);
         assertThat(records).containsExactlyElementsOf(recordsAfterChange);
     }
+
+    @Test
+    public void validateCommitAlwaysReturnsRealAddress() throws Exception {
```

Comment:
Based on the previous comment I believe we should add a test where the sequencer regresses, and we show the StreamListener can miss updates.

Reply:
Do we have any test that does an actual sequencer regression? I am only able to find TODOs like this in the code base..
`TODO: we do need a sequencer regression test, but an actual sequencer regression`